tests/ifs/sandstone_ifs.h

```diff
@@ -67,11 +67,16 @@ static ssize_t read_file(int dfd, const char *filename, char buf[static restrict
         return read_file_fd(fd, buf);
 }
 
-static int kernel_driver_is_loaded()
+static int kernel_driver_is_loaded(const char *dir)
 {
+        /* */
+        char sys_ifs_path[50];
+        strcpy(sys_ifs_path, PATH_SYS_IFS_BASE);
+        strcat(sys_ifs_path, dir);
+
         /* see if driver is loaded, otherwise try to load it */
-        int ifs0 = open(PATH_SYS_IFS_BASE "intel_ifs_0", O_DIRECTORY | O_PATH | O_CLOEXEC);
-        if (ifs0 < 0) {
+        int sys_ifs_fd = open(sys_ifs_path, O_DIRECTORY | O_PATH | O_CLOEXEC);
```

Comment:
One trick here is to use `openat` on the base dir, but I think this is fine.

Please use `strncpy` and `strncat`. Or also
```c
    int n = snprintf(sys_ifs_path, sizeof(sys_ifs_path), PATH_SYS_IFS_BASE "%s", dir);
    assert(n < sizeof(sys_ifs_path));
```

Reply:
nice solution, added!